GoogleForCreators/web-stories-wp
# Fix php notice when using customizer options to render stories.

## Context

This PR tries to resolve some of the issues found while integrating the theme support.

## Summary

    - Fixes undefined index notice in Renderer class when trying to use Customizer settings.
    - Adds title show/hide toggle customizer control for 'circles' view.
    - Replaces static options like 'order' and 'layouts' to use common utility functions.
    - Fixes issue where multiple carousels were not working on the same page.
    - Fixes issue for AMP where multiple lightbox of the same story was not working as expected.

## Relevant Technical Choices

- Uses a static value to keep updating the number of Renderer.php instances. This is to identify the number of active renderer classes for supporting multiple carousels/lightboxes.
- Could have used [spl_object_id()](https://www.php.net/manual/en/function.spl-object-id.php) for the same but it requires php 7 ( >= 7.2.0 )

## To-do

- Streamline embedding options for various mediums used i.e., shortcode, customizer, widget, block etc. As required.

## User-facing changes

- N/A

## Testing Instructions

- PHP notices issue.
   - Add theme support for the web-stories with `add_theme_support( 'web-stories' )` for your theme.
   - Render the stories in your theme's `header.php` with `new Customizer()->render_stories();`.
   - You'll see the notices on the screen if the debugger is on.
  
- Reproduction steps for carousel not working.
   - Add two or more carousel on the same page, ( shortcode, block, customizer, widget any ).
   - Notice that the first carousel is working just fine, but the second carousel and others are not working as expected.

- Lightbox not working, steps to reproduce.
   - Add two or more Web Stories block/shortcode on the same page.
   - Ensure that at least one story is common among inserted web stories.
   - Observe by clicking the common story that it won't open the lightbox.

### QA

- Follow the above-mentioned testing instructions to validate the issue and confirm that the same has been fixed.

This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---
